src/app/components/Header/HeaderSearchContext.test.tsx

```diff
@@ -0,0 +1,83 @@
+import { render } from '@testing-library/react'
+import { renderHook, act } from '@testing-library/react-hooks'
+import { mocked } from 'ts-jest/utils'
+import withAppContext from '../../utils/withAppContext'
+import useParam from '../../utils/useParam'
+import HeaderSearchContext, { HeaderSearchProvider, useHeaderSearch } from './HeaderSearchContext'
+import type { HeaderSearchContextProps } from './HeaderSearchContext'
+import HeaderSearch from './HeaderSearch'
+
+jest.mock('./HeaderSearch', () => () => <div />)
+jest.mock('../../utils/useParam')
+
+const useParamMock = mocked(useParam)
+
+describe('HeaderSearchContext', () => {
+  beforeEach(() => {
+    useParamMock.mockReturnValue([null, () => {}])
+  })
+
+  afterEach(() => {
+    useParamMock.mockClear()
+  })
+
+  it('renders', () => {
+    render(
+      withAppContext(
+        <HeaderSearchProvider>
+          <HeaderSearch />
+        </HeaderSearchProvider>,
+      ),
+    )
+  })
+
+  it('provides an empty value if the URL param is empty', () => {
+    useParamMock.mockReturnValue(['', () => {}])
+
+    const Tests = () => {
```

Comment:
You should rather write a test like this with the `renderHook` util. Then we don't need to render a test component.

```suggestion
    const { result } = renderHook(() => useHeaderSearch(), {
      wrapper: ({ children }) => <HeaderSearchProvider>{children}</HeaderSearchProvider>,
    })
    
    expect(result.current.searchInputValue).toEqual('')
```

Reply:
Aha, this is very good to know. I was a bit confused with the renderHook